frontend/src/main/java/com/google/tripmeout/frontend/servlets/PlaceParentServlet.java

```diff
@@ -0,0 +1,110 @@
+package com.google.tripmeout.frontend.servlets;
+
+import com.google.common.flogger.FluentLogger;
+import com.google.gson.Gson;
+import com.google.maps.GeoApiContext;
+import com.google.maps.PlaceDetailsRequest;
+import com.google.maps.errors.InvalidRequestException;
+import com.google.maps.model.PlaceDetails;
+import com.google.tripmeout.frontend.PlaceVisitModel;
+import com.google.tripmeout.frontend.storage.PlaceVisitStorage;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.util.List;
+import java.util.Optional;
+import java.util.regex.Pattern;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+public class PlaceParentServlet extends HttpServlet {
+  private static final FluentLogger logger = FluentLogger.forEnclosingClass();
+  private final PlaceVisitStorage placeStorage;
+  private final Gson gson;
+
+  private static final Pattern TRIP_NAME_PATTERN = Pattern.compile(".*/trips/([^/]+)/placeVisits");
+
+  private final GeoApiContext CONTEXT =
+      new GeoApiContext.Builder().apiKey("AIzaSyBbXCXC2uWv3baNmirLtqUYbFsFwCXqLV8").build();
+
+  public PlaceParentServlet(PlaceVisitStorage placeStorage, Gson gson) {
```

Comment:
Make this package private (remove the `public` modifier) and annotate with `@Inject`.